src/status_im/anon_metrics/core.cljs

```diff
@@ -0,0 +1,90 @@
+(ns status-im.anon-metrics.core
+  (:require [status-im.ethereum.json-rpc :as json-rpc]
+            [taoensso.timbre :as log]
+            [re-frame.core :as re-frame]
+            [re-frame.interceptor :refer [->interceptor]]
+            [status-im.utils.async :refer [async-periodic-exec async-periodic-stop!]]
+            [status-im.utils.platform :as platform]
+            [status-im.utils.build :as build]
+            [status-im.utils.fx :as fx]
+            [status-im.anon-metrics.transformers :as txf]))
+
+(defonce events-foyer (atom []))
+(defonce periodic-tasks-chan (atom nil))
+
+(defn onboard-events
+  "Check if there are any events in the foyer, flush them to the backend and clear foyer on-success."
+  []
+  (let [outstanding-events @events-foyer]
+    (when (seq outstanding-events)
+      (json-rpc/call {:method     "appmetrics_saveAppMetrics"
+                      :params     [outstanding-events]
+                      ;; clear the foyer when events are stored
+                      :on-success #(reset! events-foyer [])
```

Comment:
it might have new events in it, why not to clear it before call ?

Reply:
I wanted to refactor this bit, didn't realise some events would be lost. Fixed it now. The foyer is reset before the RPC.